index.html

```diff
@@ -3,47 +3,244 @@
+	{"state":"Kentucky","employment":1908.9,"region":"South"},
+	{"state":"Louisiana","employment":1977.1,"region":"South"},
+	{"state":"Maine","employment":614.6,"region":"Northeast"},
+	{"state":"Maryland","employment":2704.6,"region":"South"},
+	{"state":"Massachusetts","employment":3537.1,"region":"Northeast"},
+	{"state":"Michigan","employment":4324.9,"region":"Midwest"},
+	{"state":"Minnesota","employment":2876.2,"region":"Midwest"},
+	{"state":"Mississippi","employment":1147.1,"region":"South"},
+	{"state":"Missouri","employment":2801.2,"region":"Midwest"},
+	{"state":"Montana","employment":464.6,"region":"West"},
+	{"state":"Nebraska","employment":1017.3,"region":"Midwest"},
+	{"state":"Nevada","employment":1282.7,"region":"West"},
+	{"state":"New Hampshire","employment":664.9,"region":"Northeast"},
+	{"state":"New Jersey","employment":4073.9,"region":"Northeast"},
+	{"state":"New Mexico","employment":827.6,"region":"West"},
+	{"state":"New York","employment":9332.9,"region":"Northeast"},
+	{"state":"North Carolina","employment":4310.5,"region":"South"},
+	{"state":"North Dakota","employment":440.0,"region":"Midwest"},
+	{"state":"Ohio","employment":5491.4,"region":"Midwest"},
+	{"state":"Oklahoma","employment":1666.0,"region":"South"},
+	{"state":"Oregon","employment":1823.2,"region":"West"},
+	{"state":"Pennsylvania","employment":5894.6,"region":"Northeast"},
+	{"state":"Rhode Island","employment":490.6,"region":"Northeast"},
+	{"state":"South Carolina","employment":2039.6,"region":"South"},
+	{"state":"South Dakota","employment":432.3,"region":"Midwest"},
+	{"state":"Tennessee","employment":2951.7,"region":"South"},
+	{"state":"Texas","employment":11961.6,"region":"South"},
+	{"state":"Utah","employment":1411.7,"region":"West"},
+	{"state":"Vermont","employment":316.5,"region":"Northeast"},
+	{"state":"Virginia","employment":3919.1,"region":"South"},
+	{"state":"Washington","employment":3230.5,"region":"West"},
+	{"state":"West Virginia","employment":761.8,"region":"South"},
+	{"state":"Wisconsin","employment":2936.2,"region":"Midwest"},
+	{"state":"Wyoming","employment":284.4,"region":"West"}
+  ];
+
+  // The ALL_CAPS treatment here is signalling that this variable is a constant - it should never change.
+  var MAX_EMPLOYMENT = 16322;
+
+  var options = {
+    sorted: false,
+    filtered: false,
+    page: 0
+  };
+
+  function updateTable() {
+    var data = states.slice();
+
+    if (options.sorted) {
+      if (options.sorted === 'asc') {
+          data.sort(function (a, b) { return a.employment - b.employment; });
+        } else if (options.sorted === 'desc') {
+          data.sort(function (a, b) { return b.employment - a.employment; });
+        }
+    }    
+
+//      console.log("message here", options.sorted, options.filtered)
+
+    if (options.filtered) {
+    	if (options.filtered === 'northeast') {
+    		data = data.filter(function (d) {	return d.region === "Northeast" })
```

Comment:
Great. For next week, wonder if you could rewrite this using the `object[key]` syntax we showed in class, as opposed to the dot syntax (`object.key`) used here. The end result could be simpler, particularly if the capitalization of options.filtered matches the data.